### PR TITLE
botan-2.0.1:new recipe

### PR DESCRIPTION
**Pull request by dacianf**
I get this error when i try to build it http://pastebin.com/EDgJUhUC

### CODE REVIEW

File: dev-libs/botan/botan-2.0.1.recipe [hunk 1]
```diff
@@ -0,0 +1,71 @@
+SUMMARY="A cryptography library written in C++"
+DESCRIPTION="This is the best option for crypthography in C++"
+HOMEPAGE="https://botan.randombit.net/"
+COPYRIGHT="2000-2017 botan"
```
Comment (pulkomandy) [inline dev-libs/botan/botan-2.0.1.recipe]: Why would we reassing the copyright like this? If they say "1999-2013,2014,2015,2016,2017 The Botan Authors", then we should just use that. Reassigning to a single author is not correct.

File: dev-libs/botan/botan-2.0.1.recipe [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:g++$secondaryArchSuffix
+	cmd:make
+	cmd:python
+	"
+PATCH()
+{
+	#	sed -i -e "/^all\>/ s/-pthread //" src/build-data/cc/gcc.txt
+	sed -i -e "/^all\>/ s/-pthread -fstack-protector/-fno-stack-protector/" \
```
Comment (pulkomandy) [inline dev-libs/botan/botan-2.0.1.recipe]: Yes, the stack protection is missing and this leads to build failures (undefined references). We should patch our gcc to tell it how to support this feature.

File: dev-libs/botan/botan-2.0.1.recipe [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		src/build-data/cc/gcc.txt
+}
+
+BUILD()
+{
+	runConfigure --omit-buildspec ./configure.py \
+		--with-bzip2 --with-lzma --with-zlib \
+		--with-openssl --with-sqlite3 \
+		--destdir=/
```
Comment (pulkomandy) [inline dev-libs/botan/botan-2.0.1.recipe]: Usually running libtoolize to update libtool files will help with getting shared libraries built. Old libtool versions had a bug that prevented them to detect that shared libs can be built on Haiku.

File: dev-libs/botan/botan-2.0.1.recipe (hunk 1 above)
Comment (fbrosson) [inline dev-libs/botan/botan-2.0.1.recipe]: Well, according to the authors.txt file, Jack Lloyd is the only author. The next 6 people have just made "extensive and ongoing contributions", so my understanding is that when they contribute they give their copyright to "The Botan Authors"... which is currently Jack Lloyd.

But I guess your suggestion is OK.
BTW, do we squeeze "1999-2013,2014,2015,2016,2017" to "1999-2017"?

File: dev-libs/botan/botan-2.0.1.recipe (hunk 2 above)
Comment (fbrosson) [inline dev-libs/botan/botan-2.0.1.recipe]: OK, thanks!

File: dev-libs/botan/botan-2.0.1.recipe (hunk 1 above)
Comment (pulkomandy) [inline dev-libs/botan/botan-2.0.1.recipe]: Well, at least in France you cannot give your authorship away. You can give the copyright, but as soon as you change a line of code, you are part of the "authors".

And yes, we should squeeze the date range, of course.

File: dev-libs/botan/botan-2.0.1.recipe (hunk 3 above)
Comment (fbrosson) [inline dev-libs/botan/botan-2.0.1.recipe]: Thanks @pulkomandy for the suggestion.
I tried with a "libtoolize --force --copy --install" a few days ago, but it did not help.
I'll push it here, with the fixes I had requested in my [last review](#pullrequestreview-17802804), and maybe we can merge this PR even if the recipe does not yet build the shared lib.

**Comment by fbrosson** [comment]
@dacianf, I assumed you were busy, so I pushed the fixes and merged the recipe.

BTW, now that this PR has been merged, the branch it is based on is no longer needed.
You should therefore click "Delete branch" for this PR to get rid of an unneeded branch in your repo. Thanks!
GitHub help: [Deleting unused branches](https://help.github.com/articles/deleting-unused-branches/)

**Pull request merged**